tfctl/main.go

```diff
@@ -0,0 +1,110 @@
+package tfctl
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/fluxcd/pkg/ssa"
+	"github.com/spf13/viper"
+	infrav1 "github.com/weaveworks/tf-controller/api/v1alpha1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/rest"
+
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"sigs.k8s.io/cli-utils/pkg/kstatus/polling"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	repo = "weaveworks/tf-controller"
+)
+
+// CLI is the main struct for the tfctl command line tool
+type CLI struct {
+	restConfig *rest.Config
+	client     client.Client
+	namespace  string
+	terraform  string
+	build      string
+	release    string
+}
+
+// New returns a new CLI instance
+func New(build, release string) *CLI {
+	return &CLI{build: build, release: release}
+}
+
+// Init initializes the CLI instance for a given kubeconfig, namespace and terraform binary
+func (c *CLI) Init(config *viper.Viper) error {
+	var kubeconfigArgs = genericclioptions.NewConfigFlags(false)
```

Comment:
I think you can add these to the underlying flagset in Cobra which would save you duplicating all the flags in the ConfigFlags?

https://pkg.go.dev/k8s.io/cli-runtime/pkg/genericclioptions#ConfigFlags.AddFlags